### DISCUSSION TITLE
Silencing Unknown class 'SPUStandardUpdaterController' warnings

### DISCUSSION THREAD

**Discussion by massimobio** [sparkle-project/Sparkle#2509] [Q&A] [answered]
My Mac app has two targets. One for the Mac App Store and one for direct distribution which uses the Sparkle library.
I don't include Sparkle in the MAS target so I always see this warning in Xcode:

<img width="382" alt="Screenshot 2024-02-18 at 11 33 39 PM" src="https://github.com/sparkle-project/Sparkle/assets/240599/4e9853e0-38c2-493e-8d59-f046208e224c">

and this warning in the console when running the app:
```
Unknown class 'SPUStandardUpdaterController', using 'NSObject' instead. Encountered in Interface Builder file at path '.../MainMenu.nib'.
Could not connect action. Target class 'NSObject' does not respond to '-checkForUpdates:'.
```

I attempted to silence this warning by creating a placeholder class for the MAS target:
```swift
class SPUStandardUpdaterController: NSObject {
    @IBAction func check(forUpdates sender: Any) {}
}
```
but it didn't work.

Short of having separate MainMenu.xib files for each target (not great for maintainability), is there some other way to silence these warnings?

**Comment by zorgiepoo** [comment, marked as answer]
Probably best to instantiate the controller programmatically then instead of in the main nib. See https://sparkle-project.org/documentation/programmatic-setup/ it should only be a few lines of code.
